File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = 0;
        for (int i=0; i<nums.size(); i++) {
            n = nums[i];
            if (i != n) {
                return i;
            }
        }
        return n+1;
    }
};

File: 268-missing-number/268-missing-number.py
class Solution:
    # O(1) space, O(n) time
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        size = (n * (n + 1)) // 2
        return int(size - sum(nums))
    # bad, nlog n
    """
    def missingNumber(self, nums: List[int]) -> int:
        nums.sort()
        for i, n in enumerate(nums):
            if i != n:
                return i
        return n+1
    """